zrml/prediction-markets/src/lib.rs

```diff
@@ -842,21 +865,37 @@ mod pallet {
                 );
 
                 let mut should_check_origin = false;
+                //NOTE: Saturating operation in following block may saturate to u32::MAX value
+                //      but that will be the case after thousands of years time. So it is fine.
                 match market.period {
                     MarketPeriod::Block(ref range) => {
-                        if current_block
-                            <= range.end.saturating_add(T::ReportingPeriod::get().into())
-                        {
+                        let grace_period_end =
+                            range.end.saturating_add(market.deadlines.grace_period);
+                        ensure!(
+                            grace_period_end <= current_block,
+                            Error::<T>::NotAllowedToReportYet
+                        );
```

Comment:
as per https://github.com/zeitgeistpm/zeitgeist/pull/732#discussion_r955324146 
say end is at 4 , grace_period = 3 and current_block is 7 then for block number 7 it should allow to report but with your proposed change it will error. right?

Reply:
It's okay as it is 👍